src/panels/lovelace/editor/types.ts

```diff
@@ -168,7 +168,7 @@ export const entitiesConfigStruct = union([
   object({
     entity: EntityId,
     name: optional(string()),
-    icon: optional(Icon),
+    icon: optional(string()),
```

Comment:
why?

Reply:
Because idk why this was ever a thing :). If I am editing the icon. and I type anything but `mdi:<Icon-name>` then if is a struct error

Reply:
Oh, so it switches to the YAML editor?

Reply:
Yea...

Reply:
Same with the Entity ID. With this PR. I proposing using generic struct types and let the card error/warn about bad entity ids. Otherwise, trying to type an entity id into the code editor makes it jump because you haven't finished your input